scene/resources/particle_process_material.cpp

```diff
@@ -421,396 +502,491 @@ void ParticleProcessMaterial::_update_shader() {
+	}
+	if (emission_shape == EMISSION_SHAPE_RING) {
+		code += "		\n";
+		code += "		float ring_spawn_angle = rand_from_seed(alt_seed) * 2.0 * pi;\n";
+		code += "		float ring_random_radius = rand_from_seed(alt_seed) * (emission_ring_radius - emission_ring_inner_radius) + emission_ring_inner_radius;\n";
+		code += "		vec3 axis = normalize(emission_ring_axis);\n";
+		code += "		vec3 ortho_axis = vec3(0.0);\n";
+		code += "		if (axis == vec3(1.0, 0.0, 0.0)) {\n";
+		code += "			ortho_axis = cross(axis, vec3(0.0, 1.0, 0.0));\n";
+		code += "		} else {\n";
+		code += " 			ortho_axis = cross(axis, vec3(1.0, 0.0, 0.0));\n";
+		code += "		}\n";
+		code += "		ortho_axis = normalize(ortho_axis);\n";
+		code += "		float s = sin(ring_spawn_angle);\n";
+		code += "		float c = cos(ring_spawn_angle);\n";
+		code += "		float oc = 1.0 - c;\n";
+		code += "		ortho_axis = mat3(\n";
+		code += "			vec3(c + axis.x * axis.x * oc, axis.x * axis.y * oc - axis.z * s, axis.x * axis.z *oc + axis.y * s),\n";
+		code += "			vec3(axis.x * axis.y * oc + s * axis.z, c + axis.y * axis.y * oc, axis.y * axis.z * oc - axis.x * s),\n";
+		code += "			vec3(axis.z * axis.x * oc - axis.y * s, axis.z * axis.y * oc + axis.x * s, c + axis.z * axis.z * oc)\n";
+		code += "			) * ortho_axis;\n";
+		code += "		ortho_axis = normalize(ortho_axis);\n";
+		code += "		pos = ortho_axis * ring_random_radius + (rand_from_seed(alt_seed) * emission_ring_height - emission_ring_height / 2.0) * axis;\n";
+	}
+
+	code += "	return pos * emission_shape_scale + emission_shape_offset;\n";
+	code += "}\n";
+	code += "\n";
+	if (tex_parameters[PARAM_ORBIT_VELOCITY].is_valid() || particle_flags[PARTICLE_FLAG_DISABLE_Z]) {
+		code += "vec3 process_orbit_displacement(DynamicsParameters param, float lifetime, inout uint alt_seed, mat4 transform, mat4 emission_transform,float delta, float total_lifetime){\n";
+		// No reason to run all these expensive calculation below if we have no orbit velocity
+		// HOWEVER
+		// May be a bad idea for fps consistency?
+		code += "if(abs(param.orbit_velocity) < 0.01){ return vec3(0.0);}\n";
+		code += "\n";
+		code += "	vec3 displacement = vec3(0.);\n";
+		code += "	float pi = 3.14159;\n";
+		code += "	float degree_to_rad = pi / 180.0;\n";
+		if (particle_flags[PARTICLE_FLAG_DISABLE_Z]) {
+			code += "	float orbit_amount = param.orbit_velocity;\n";
+
+			if (tex_parameters[PARAM_ORBIT_VELOCITY].is_valid()) {
+				code += "   orbit_amount *= texture(orbit_velocity_curve, vec2(lifetime)).rgb;\n";
+			}
+			code += "	if (orbit_amount != 0.0) {\n";
+			code += "       vec3 org = EMISSION_TRANSFORM[3].xyz;\n";
+			code += "       vec3 diff = pos - org;\n";
+			code += "	     float ang = orbit_amount * delta * pi * 2.0;\n";
+			code += "	     mat2 rot = mat2(vec2(cos(ang), -sin(ang)), vec2(sin(ang), cos(ang)));\n";
+			code += "	     displacement.xy -= diff.xy;\n";
+			code += "        displacemen.xy += rot * diff.xy;\n";
```

Comment:
```suggestion
			code += "        displacement.xy += rot * diff.xy;\n";
```

Reply:
Hey, is this a typo that prevents the shader from functioning? Otherwise I think it needs to change also in the variable definition and all other usages.

Reply:
Probably, but I'm not sure, as I've simply quickly skimmed through the changes on my phone. Yet, this variable is declared at line 662 and used at line 676 as `displacement`, so it'd be quite peculiar if this code would work properly while referring an undefined variable (`displacemen`).

Reply:
Oooh thank you! I'll double check

Reply:
Thanks it was indeed a typo!